prometheus-to-sd/translator/translator.go

```diff
@@ -42,28 +43,59 @@ var supportedMetricTypes = map[dto.MetricType]bool{
 
 const falseValueEpsilon = 0.001
 
-// TranslatePrometheusToStackdriver translates metrics in Prometheus format to Stackdriver format.
-func TranslatePrometheusToStackdriver(config *config.CommonConfig,
-	whitelisted []string,
-	metrics map[string]*dto.MetricFamily,
-	cache *MetricDescriptorCache) []*v3.TimeSeries {
+type timeSeriesBuilder struct {
```

Comment:
I would say that is a bit unusual to return private struct, but at the same time to have all methods public. For example that means timeSeriesBuilder couldn't be passed into any function.

Reply:
It can be passed as an interface, if needed. Making the struct private ensures that it's created only using the New* function. See http://www.golangpatterns.info/object-oriented/constructors